addons/mail/static/src/models/attachment_card/attachment_card.js

```diff
@@ -15,14 +15,7 @@ registerModel({
             if (!this.attachment || !this.attachment.isViewable) {
                 return;
             }
-            this.messaging.dialogManager.update({
-                dialogs: insert({
-                    attachmentViewer: insertAndReplace({
-                        attachment: replace(this.attachment),
-                        attachmentList: replace(this.attachmentList),
-                    }),
-                }),
-            });
+            this.update({ attachmentViewDialog: insertAndReplace() });
```

Comment:
This should still "link" the "current attachment" as the one that is displayed by the viewer when opening it.